components/job-orchestration/job_orchestration/scheduler/query/query_scheduler.py

```diff
@@ -542,10 +547,51 @@ def handle_pending_query_jobs(
                         logger.error(f"Failed to set job {job_id} as failed")
                     continue
 
+                # Note: the following two if blocks should not be reordered.
+                # The second if statement only check if any chunk of the specific file split is extracted, but
+                # it does not guarantee all chunks to be ready because the extraction job could still be running.
+                # However, if there is no running extraction job for the file split, then a single chunk being
+                # available guarantees that all chunks are extracted.
+                if file_split_id in active_extraction_file_splits:
+                    active_extraction_file_splits[file_split_id].append(job_id)
+                    logger.info(
+                        f"Split {file_split_id} is being extracted, mark job {job_id} as running"
+                    )
+                    if not set_job_or_task_status(
```

Comment:
imo, this `if not set_job_or_task_status error` is quite annoying to write and repeated several places in our code. I want to wrap it with a method, but not sure if we want to put in a separate PR.